### PR TITLE
Bugzilla 2483

### PR DESCRIPTION
**Pull request by joochlee**
OAS for BZ2483 - Push Notification

### CODE REVIEW

File: swagger2.0/oic.r.pushreceiver.swagger.json [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+      "name": "receivers",
+      "required": true,
+      "schema": {
+        "$ref": "#/definitions/post-rw-request"
+      },
+      "x-example": {
+        "receivers": [
+          { 
+            "receiveruri": "/mylocaltargeturifordontcare", 
+            "rts": []
```
Comment (WAvdBeek) [inline swagger2.0/oic.r.pushreceiver.swagger.json:167]: should there be a value here (as example)?

File: swagger2.0/oic.r.notificationselector-pushproxy.swagger.json [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+      "get": {
+        "description": "Resource that defines the selector for Push Notifications",
+        "parameters": [
+          {"$ref": "#/parameters/interface-rw"}
+        ],
+        "responses": {
+          "200": {
+            "description" : "",
+            "x-example": {
+              "phref": "/myAirquality",
```
Comment (WAvdBeek) [inline swagger2.0/oic.r.notificationselector-pushproxy.swagger.json:33]: please add rt in the get examples everywhere

File: swagger2.0/oic.r.pushconfiguration.swagger.json [hunk 3]
```diff
@@ -0,0 +1,532 @@
...
+      "get": {
+        "description": "Collection of oic.r.notificationselector with associated push proxies.\nAllows a Server to be configured with one or more Push Notification destinations.\n",
+        "parameters": [
+          {"$ref": "#/parameters/interface-ll"}
+        ],
+        "responses": {
+          "200": {
+            "description" : "",
+            "x-example": [
+              {"href": "/pushconfig/1", "rt": ["oic.r.notificationseltor","oic.r.pushproxy"], "if": ["oic.if.rw","oic.if.baseline"], "eps": [{"ep": "coaps://[fe80::b1d6]:1122"}]},
```
Comment (WAvdBeek) [inline swagger2.0/oic.r.pushconfiguration.swagger.json]: typo: notificationselector

File: swagger2.0/oic.r.notificationselector-pushproxy.swagger.json [hunk 4]
```diff
@@ -0,0 +1,334 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Notification Selector",
+    "version": "2019-03-25",
```
Comment (WAvdBeek) [inline swagger2.0/oic.r.notificationselector-pushproxy.swagger.json]: update the dates

File: swagger2.0/oic.r.notificationselector-pushproxy.swagger.json [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+    "license": {
+      "name": "OCF Data Model License",
+      "url": "https://github.com/openconnectivityfoundation/core/blob/e28a9e0a92e17042ba3e83661e4c0fbce8bdc4ba/LICENSE.md",
```
Comment (WAvdBeek) [inline swagger2.0/oic.r.notificationselector-pushproxy.swagger.json:8]: update the copyright to 2022

File: swagger2.0/oic.r.pushreceiver.swagger.json (hunk 1 above)
Comment (joochlee) [reply swagger2.0/oic.r.pushreceiver.swagger.json:167]: do you mean `"rts"`? 
nop, `"rts"` can be empty (`"minItems": 0`). (see "#/definitions/receivers")

If `"rts"` is empty, any type of Resource can be posted to the corresponding `"receiveruti"`

File: swagger2.0/oic.r.notificationselector-pushproxy.swagger.json (hunk 2 above)
Comment (joochlee) [reply swagger2.0/oic.r.notificationselector-pushproxy.swagger.json:33]: I'm little bit confused here.
According to the core spec, difference b/w `oic.if.baseline` and other interface such as `oic.if.rw` is that baseline shall include common Properties. 
Current iotiviti-lite returns Properties except for common Properties only against RETRIEVE through `oic.if.rw` interface.
But there is no specific text limiting non-common Properties when `oic.if.rw` interface is applied in the spec. 

Which is the correct operation ?

File: swagger2.0/oic.r.pushconfiguration.swagger.json (hunk 3 above)
Comment (joochlee) [reply swagger2.0/oic.r.pushconfiguration.swagger.json]: fixed

File: swagger2.0/oic.r.notificationselector-pushproxy.swagger.json (hunk 4 above)
Comment (joochlee) [reply swagger2.0/oic.r.notificationselector-pushproxy.swagger.json]: fixed

File: swagger2.0/oic.r.notificationselector-pushproxy.swagger.json (hunk 5 above)
Comment (joochlee) [reply swagger2.0/oic.r.notificationselector-pushproxy.swagger.json:8]: fixed

File: swagger2.0/oic.r.notificationselector-pushproxy.swagger.json (hunk 2 above)
Comment (WAvdBeek) [reply swagger2.0/oic.r.notificationselector-pushproxy.swagger.json:33]: ok leave it like this.

**Review by WAvdBeek — APPROVED** [review]

**Pull request merged**